js/piwik.js

```diff
@@ -2672,7 +2672,7 @@ if (typeof window.Matomo !== 'object') {
                     url = 'http://' + url;
                 }
 
-                parser.href = content.toAbsoluteUrl(url);
+                parser.href = encodeURI(content.toAbsoluteUrl(url)); // Sanitization and encoding
```

Comment:
Wouldn't that cause a double encoding? If you e.g. have a url like `https://matomo.org/path%20test/?x=yt`, this code would then result in `/path%2520test/` as pathname.
Also I guess some automatic encoding also happens there. If you use `parser.href = 'https://matomo.org/path test/?x=y';` (without calling `encodeURI`) the path name will return `/path%20test/`

Reply:
You are correct. Since I didn't have any cases with paths with special characters, I didn't pay attention to that. I made a new version. Simpler, but it passes the Checkmarx AST scanner.

Reply:
@thiagoalvescosta how would that handle e.g. `mailto:user@example.com?subject=hello` or `tel:123456789` links?

Reply:
From what I understood, you just wanted the path of a given base URL. But if tel: or mailto: are important, I changed it and it is passing Checkmarx AST scanner successfully.